src/c_vec.rs

```diff
@@ -273,12 +228,19 @@ impl<T> CSlice<T> {
     pub fn is_empty(&self) -> bool { self.len() == 0 }
 }
 
-impl<T> AsSlice<T> for CSlice<T> {
-    /// View the stored data as a slice.
-    fn as_slice<'a>(&'a self) -> &'a [T] {
```

Comment:
I fixed that in #6. However, I really need to think about how improving all this because it's becoming a mess...